### PR TITLE
[WIP] 001 Script for requirement [GENIVI] [Policies] External UCS: informing HMI

### PR DESCRIPTION
**Pull request by maria-mih**
@istoimenova Please review 
001_ATF_P_GetListofPermissions_with_appID_all_ecs_in_bound_EntitySatus_ON.lua 
Testing coverage is under review in #1165 
Applicable json available at files/jsons/Policies/Related_HMI_API/OnAppPermissionConsent_ptu.json

### CODE REVIEW

File: test_scripts/Polices/External_UCS_Informing_HMI/001_ATF_P_GetListofPermissions_with_appID_all_ecs_in_bound_EntitySatus_ON.lua [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+function Test:Precondition_trigger_getting_device_consent()
+  testCasesForPolicyTable:trigger_getting_device_consent(self, config.application1.registerAppInterfaceParams.appName, config.deviceMAC)
+end
+
+function Test:Precondition_PTU_and_OnAppPermissionConsent_AllParams_Valid()
+  testCasesForPolicyTable:flow_SUCCEESS_EXTERNAL_PROPRIETARY(self)
+  EXPECT_HMINOTIFICATION("SDL.OnAppPermissionChanged",{ appID = self.applications[config.application1.registerAppInterfaceParams.appName]})
+  :Do(function(_,data)
+      if (data.result.appPermissionsConsentNeeded == true) then
```
Comment (istoimenova) [inline test_scripts/Polices/External_UCS_Informing_HMI/001_ATF_P_GetListofPermissions_with_appID_all_ecs_in_bound_EntitySatus_ON.lua]: @maria-mih, parameter 'data' doesn't have element result, you should use data.params.appPermissionsConsentNeeded

File: test_scripts/Polices/External_UCS_Informing_HMI/001_ATF_P_GetListofPermissions_with_appID_all_ecs_in_bound_EntitySatus_ON.lua [hunk 2]
```diff
@@ -0,0 +1,127 @@
+---------------------------------------------------------------------------------------------
```
Comment (istoimenova) [inline test_scripts/Polices/External_UCS_Informing_HMI/001_ATF_P_GetListofPermissions_with_appID_all_ecs_in_bound_EntitySatus_ON.lua]: @maria-mih, please clarify with BA should SDL send OnAppPermissionChanged when:

Precondition:
1. Application is listed in sdl_preloaded_pt.json, device is not consented. 
2. Application has assigned groups that require user consent.
3. Start SDL and HMI, 
4. Register the application(it is listed in preloaded file.)
5. Activate application.
Step:
1. PTU is triggered to get consent of device. ("isPermissionsConsentNeeded":true)

Question: Should SDL send OnAppPermissionChanged to request user consent OR
OnAppPermissionChanged should be sent only after PTU.
I didn't manage to find applicable requirement for this situation.

Note: if SDL should send OnAppPermissionChanged after device consent, script should be updated .

**Comment by maria-mih** [comment]
@istoimenova Please start second review iteration on commit 3c7425d3617e640914c323d15ffb98a67ba1de27

File: test_scripts/Polices/External_UCS_Informing_HMI/001_ATF_P_GetListofPermissions_with_appID_all_ecs_in_bound_EntitySatus_ON.lua [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+--
+-- Description:
+-- For Genivi applicable ONLY for 'EXTERNAL_PROPRIETARY' Polcies
+-- Upon GetListOfPermissions_request, SDL must inform "externalConsentStatus" setting to HMI
+--
+-- 1. Used preconditions
+-- SDL is built with External_Proprietary flag
+-- SDL and HMI are running
+-- Application is registered and activated
+-- PTU has passed successfully
```
Comment (istoimenova) [inline test_scripts/Polices/External_UCS_Informing_HMI/001_ATF_P_GetListofPermissions_with_appID_all_ecs_in_bound_EntitySatus_ON.lua]: @maria-mih, please add additional information that ptu file is updated and which functional groups are assigned to application.

File: test_scripts/Polices/External_UCS_Informing_HMI/001_ATF_P_GetListofPermissions_with_appID_all_ecs_in_bound_EntitySatus_ON.lua [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+end
+
+function Test:Precondition_PTU_and_OnAppPermissionConsent_AllParams_Valid()
+  local app_id = config.application1.registerAppInterfaceParams.appID 
+  local device_id = config.deviceMAC 
+  local hmi_app_id = self.applications[config.application1.registerAppInterfaceParams.appName]
+  local ptu_file_path = "files/jsons/Policies/Related_HMI_API/"
+  local ptu_file = "OnAppPermissionConsent_ptu.json"
+  
+  testCasesForPolicyTable:flow_SUCCEESS_EXTERNAL_PROPRIETARY(self, app_id, device_id, hmi_app_id, ptu_file_path, nil, ptu_file)
```
Comment (istoimenova) [inline test_scripts/Polices/External_UCS_Informing_HMI/001_ATF_P_GetListofPermissions_with_appID_all_ecs_in_bound_EntitySatus_ON.lua]: @maria-mih, as you are using default parameters in flow_SUCCEESS_EXTERNAL_PROPRIETARY for app_id, device_id,hmi_app_id you may not declare them and to live them as nil

File: test_scripts/Polices/External_UCS_Informing_HMI/001_ATF_P_GetListofPermissions_with_appID_all_ecs_in_bound_EntitySatus_ON.lua [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+--[[ Required Shared libraries ]]
+local commonFunctions = require ('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require('user_modules/shared_testcases/commonSteps')
+local commonPreconditions = require('user_modules/shared_testcases/commonPreconditions')
+local testCasesForPolicyTable = require('user_modules/shared_testcases/testCasesForPolicyTable')
+
+--[[ General Precondition before ATF start ]]
+commonSteps:DeleteLogsFileAndPolicyTable()
+testCasesForPolicyTable:Precondition_updatePolicy_By_overwriting_preloaded_pt("files/jsons/Policies/Related_HMI_API/OnAppPermissionConsent.json")
```
Comment (maria-mih) [inline test_scripts/Polices/External_UCS_Informing_HMI/001_ATF_P_GetListofPermissions_with_appID_all_ecs_in_bound_EntitySatus_ON.lua]: @istoimenova removed

File: test_scripts/Polices/External_UCS_Informing_HMI/001_ATF_P_GetListofPermissions_with_appID_all_ecs_in_bound_EntitySatus_ON.lua (hunk 3 above)
Comment (maria-mih) [inline test_scripts/Polices/External_UCS_Informing_HMI/001_ATF_P_GetListofPermissions_with_appID_all_ecs_in_bound_EntitySatus_ON.lua]: @istoimenova Information added

File: test_scripts/Polices/External_UCS_Informing_HMI/001_ATF_P_GetListofPermissions_with_appID_all_ecs_in_bound_EntitySatus_ON.lua (hunk 4 above)
Comment (maria-mih) [inline test_scripts/Polices/External_UCS_Informing_HMI/001_ATF_P_GetListofPermissions_with_appID_all_ecs_in_bound_EntitySatus_ON.lua]: @istoimenova Parameters are left as nil

File: test_scripts/Polices/External_UCS_Informing_HMI/001_ATF_P_GetListofPermissions_with_appID_all_ecs_in_bound_EntitySatus_ON.lua [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+      externalConsentStatus = {
+        {entityType = 13, entityID = 113, status = "ON"}
+      }
+    })
+end
+
+--[[ Postconditions ]]
+commonFunctions:newTestCasesGroup("Postconditions")
+
+function Test.Postcondition_Restore_preloaded_file()
```
Comment (maria-mih) [inline test_scripts/Polices/External_UCS_Informing_HMI/001_ATF_P_GetListofPermissions_with_appID_all_ecs_in_bound_EntitySatus_ON.lua]: @istoimenova Removed

**Pull request merged**